### PR TITLE
Add hack/verify-test-owners.sh to ensure tests always have owners.

### PR DESCRIPTION
**Pull request by rmmh**
This ensures that new tests or changed tests are assigned appropriate owners.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35650)

### CODE REVIEW

**Comment by rmmh** [comment]
Note: this should fail the verification step, as a check that the new verify script is actually running.

**Comment by fejta** [comment]
Cool! Please self LGTM once this is ready (I suspect you'll need to update tests)

File: hack/verify-test-owners.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+set -o errexit
+set -o nounset
+set -o pipefail
+
+export KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+cd "${KUBE_ROOT}"
+if ! hack/update_owners.py --check; then
+    echo 'Run ./hack/update_owners.py to fix it'
```
Comment (ixdy) [inline hack/verify-test-owners.sh]: this should probably exit nonzero?

Comment (rmmh) [inline hack/verify-test-owners.sh]: Good catch!

**Comment by fejta** [comment]
Still LGTM

**Comment by rmmh** [comment]
Bumping priority because this needs to be changed every time a test name changes.

**Comment by fejta** [comment]
Ack, gubernator

**Comment by erictune** [comment]
Was there discussion about this change anywhere?  I would have liked to offer some feedback before this was decided.

**Comment by fejta** [comment]
Answered here: https://groups.google.com/d/topic/kubernetes-wg-contribex/HVjGsZSDb9w/discussion

**Comment by thockin** [comment]
I've seen a bunch of conflicts on this file, causing need to rebase.  Can we possibly de-centralize this?

**Comment by rmmh** [comment]
PR #36075 should fix the phantom conflicts.

**Pull request merged**